### ISSUE TITLE
Implement <details>

### ISSUE THREAD

**Issue by nacnudus** [ukgovdatascience/govdown#7]
https://design-system.service.gov.uk/components/details/

**Comment by TimTaylor** [comment]
Does the following look like how you were thinking of implementing details?  If so I can raise a pull request.
```
-- look for details
      v,i = el.classes:find("details")
      if i ~= nil then
        el.classes[i] = nil
        
        local html
        local tmp = List:new{}
        
        table.insert(tmp, pandoc.RawBlock('html', '<details class="govuk-details">'))
        
        for _, block in pairs(el.content) do
          if block.t == "Header" then
            html = 
              '<summary class="govuk-details__summary">' ..
              '<span class="govuk-details__summary-text">'..
              pandoc.utils.stringify(block.content) ..
              '</span>'..
              '</summary>'
            table.insert(tmp, pandoc.RawBlock('html', html))
          else 
            html = 
              '<div class="govuk-details__text">' ..
              pandoc.utils.stringify(block.content) ..
              '</div>'
            table.insert(tmp, pandoc.RawBlock('html', html))
          end
        end
        table.insert(tmp, pandoc.RawBlock('html', '</details>'))
      
        return tmp
      end
```

**Comment by TimTaylor** [comment]
In the above I've assumed any level header can be used for summary and anything else for text.

**Comment by nacnudus** [comment]
Nice work, thank you!  You can check the header level with `if block.level == 1` once you know the block is a header.

There's a slightly bigger snag here though, which is the stringify step.  It reduces all subsequent markdown to plain text.  You could do something like

```lua
local sections = List:new{}
```
before the `if` and then
```lua
  else
    table.insert(sections, block)
```
to return blocks unmodified.  That's how `tabset` works.

Alternatively (I think is better and easier), the summary text could be from an attribute, e.g.
```md
::: {.details summary="Expand to see more"}
Too much information
:::
```
That wouldn't interfere with headers at all, and you wouldn't even need to iterate over the blocks in `el.content`.  It would work more like `unbulleted-list`, which sets up a couple of initial `RawBlock`s, then inserts each of the blocks in `el.content` and then closes with the terminating `RawBlock`s.

~~Separately, it would be good to filter HTML `<details>` and `<summary>` tags, which would mean packages like https://github.com/yonicd/details would still work.~~ I think the only way to do that would be to re-run pandoc over the generated HTML file so that it interprets the HTML as HTML, not as text.

**Comment by TimTaylor** [comment]
The following uses your suggestion of an attribute for summary ~~test~~ and seems to maintain the markdown.  
```
 -- look for details
      v,i = el.classes:find("details")
      if i ~= nil then
        el.classes[i] = nil
        
        local html
        local tmp = List:new{}
        
        html = 
          '<details class="govuk-details">' ..
          '<summary class="govuk-details__summary">' ..
          '<span class="govuk-details__summary-text">'..
          pandoc.utils.stringify(el.attributes["summary"]) ..
          '</span>'..
          '</summary>' ..
          '<div class="govuk-details__text">'
        
        table.insert(tmp, pandoc.RawBlock('html', html))
        
        for _, block in pairs(el.content) do
          table.insert(tmp, block)
        end
        table.insert(tmp, pandoc.RawBlock('html', '</div>'))
        table.insert(tmp, pandoc.RawBlock('html', '</details>'))
      
        return tmp
      end
```
Is this more what you were thinking?

**Comment by nacnudus** [comment]
That'll do nicely!

One pedantic quibble, I'd prefer `tmp` to be called `res` for consistency.

Looking forward to your PR.  Please include an example somewhere in `index.Rmd` and a bullet in `NEWS.md`.  I'll quickly do a version 0.1.0 release, and then if you could make the changes on top of that, that would be best.

**Comment by nacnudus** [comment]
Closed by #12
